test_conformance/spirv_new/types.hpp

```diff
@@ -89,6 +89,13 @@ T genrandReal(RandomSeed &seed)
     return genrand_real1(seed);
 }
 
+// Longer-term this could be refactored out and replace random_float():
+template <typename T> T genrandReal_range(T low, T high, RandomSeed &seed)
+{
+    T t = genrand_real1(seed);
+    return (1.0 - t) * low + t * high;
```

Comment:
Floating point math is really difficult. If this had been `t * low + (1.0 - t) * high`, I know this could easily have resulted in random values outside of `[low, high]`. For `(1.0 - t) * low + t * high`, as long as `low < high`, I have not seen any values for which that can happen, but if this truly cannot happen, can you add a comment explaining why?

But I am actually not sure a uniform distribution is what we want here. Suppose we have a range of `[1, 4]`. There are as many floating point values in `[1, 2)` as there are in `[2, 4)`. This means individual problematic cases in `[1, 2)` only have half the probability of being tested of those in `[2, 4)`. At the same time, if there is a whole range of values that is problematic, you do want a uniform distribution. So I do not know what is best here.